import "ads/bsyeti/big_rt/lib/consuming_system/config/config.proto";
import "ads/bsyeti/big_rt/lib/processing/shard_processor/stateless/config/stateless_shard_processor_config.proto";
import "ads/bsyeti/big_rt/lib/supplier/config/supplier_config.proto";

import "ads/bsyeti/libs/profiling/solomon/proto/config.proto";
import "ads/bsyeti/libs/ytex/http/proto/config.proto";
import "ads/bsyeti/libs/ytex/logging/proto/config.proto";

import "library/cpp/getoptpb/proto/confoption.proto";

package NAlice.NWonderlogs;

message TStaticTableCreatorConfig {
    optional NYTEx.NLogging.TConfig Logs = 14;
    optional NYTEx.NHttp.TServerConfig HttpServer = 15;
    optional NBSYeti.NProfiling.TExporterConfig SolomonExporter = 16;

    repeated NBigRT.TSupplierConfig Suppliers = 7;
    optional NBigRT.TConsumingSystemConfig ConsumingSystem = 8;
    optional NBigRT.TStatelessShardProcessorConfig StatelessShardProcessorConfig = 11;

    message TProcessorConfig {
        optional string Cluster = 1;
        optional string OutputDir = 2;
        optional uint64 GranularityMinutes = 3;
    }

    optional TProcessorConfig ProcessorConfig = 10;
    optional uint64 MaxInflightBytes = 12 [default = 1000000];
}
